src/presenter/film-presenter.js

```diff
@@ -1,71 +1,61 @@
-  #isOpened = false;
+  #isPopupOpen = false;
 
   constructor(container) {
     this.#container = container;
   }
 
+  init = (film, comments) =>{
+    this.#renderFilm(film, comments);
+  };
 
-  init = ( film, comments) => {
-    // const filmCardView = new FilmCardView(film);
-    // render(filmCardView, this.#container);
-
-    // this.popup = new PopupView(film, comments);
-
-    // filmCardView.element.addEventListener('click', ()=>{
-    //   render(
-    //     this.popup,
-    //     document.querySelector('.footer'),
-    //     RenderPosition.AFTEREND
-    //   );
-    // });
-
+  #renderFilm(film,comments) {
+    const filmCardView = new FilmCardView(film);
+    render(filmCardView, this.#container);
 
-    const cardComponent = new FilmCardView(film, comments);
-    const hideDetailsComponent = () => {
-      this.#popup.element.remove();
-      document.body.classList.remove('hide-overflow');
-    };
+    filmCardView.element.querySelector('.film-card__link').addEventListener('click', () => {
+      this.#renderFilmPopup(film,comments);
+    });
+  }
 
-    const onWindowKeydown = (evt) => {
-      if (isEscapeKey(evt)) {
-        evt.preventDefault();
-        hideDetailsComponent();
-        window.removeEventListener('keydown', onWindowKeydown);
-      }
-    };
+  #renderFilmPopup = (film,comments) => {
+    this.popup = new PopupView(film, comments);
 
-    const viewDetailsComponent = () => {
-      if (document.body.querySelector('.film-details')) {
-        return;
-      }
-      if (!this.#popup) {
-        this.#popup = new PopupView(film, comments);
-        render(this.#popup, document.querySelector('.footer'),
-          RenderPosition.AFTEREND);
-        this.#popup.element.querySelector('.film-details__close-btn').addEventListener('click', () => {
-          hideDetailsComponent();
-          window.removeEventListener('keydown', onWindowKeydown);
-        });
-      }
+    if (!this.#popup) {
+      this.#isPopupOpen = true;
+      this.#popup = new PopupView(film, comments);
+      render(this.#popup, document.querySelector('.footer'),
+        RenderPosition.AFTEREND);
+      document.body.classList.add('hide-overflow');
 
+      const popupCloseButton = this.#popup.element.querySelector('.film-details__close-btn');
+      popupCloseButton.addEventListener('click', ()=>{
+        this.#hideFilmPopup();
+      });
       document.body.classList.add('hide-overflow');
       document.body.appendChild(this.#popup.element);
-      window.addEventListener('keydown', onWindowKeydown);
-    };
+      window.addEventListener('keydown', this.#onWindowKeydown);
+    }
+  };
 
-    cardComponent.element.querySelector('.film-card__link').addEventListener('click', () => {
-      viewDetailsComponent();
-    });
-    render(cardComponent, this.#container);
+  #hideFilmPopup(){
+    if (this.#isPopupOpen) {
+      this.#popup.element.remove();
+      this.#popup = null;
+      document.body.classList.remove('hide-overflow');
```

Comment:
Надо удалять при любом закрытии keydown на window (К слову это и причина почему мы просто не можем напримую закрыть соседние презентеры удалением их из ДОМ, например, потому что мы должны для закрытия у всех выполнить какую-то определенную логику закрытия (очистить глобальные слушатели, откатить изменения, применить изменеия и т.д. что угодно), т.е. таким образом для закрытия остальных попапов при открытии другого мы должны будем у родительского презентера вызвать метод закрытия всех, и у каждого презентера фильма вызывать этот `hideFilmPopup()`)
```suggestion
      document.body.classList.remove('hide-overflow');
      window.removeEventListener('keydown', this.#onWindowKeydown);
```

Reply:
Таким образом этот метод нам нужен будет не приватным, а публичным, решетку в начале надо будет удалить, скорее всего, либо доп апстракцию в  виде публичного метода добавить, там дальше видно будет, посмотришь